### PR TITLE
upgrade(eslint): ^1.10.3 -> ^2.2.0 + update related deps

### PR DESCRIPTION
**Pull request by davezuko**

### CODE REVIEW

**Comment by jhabdas** [comment]
:boat:

File: frontend/_base.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ module.exports = {
     'browser': true,
   },
   'extends': [
-    'defaults/rules/react/airbnb',
+    'airbnb/rules/react',
```
Comment (davezuko) [inline frontend/_base.js]: defaults is not up to date, and we're just using airbnb anyways.

File: frontend/_base.js [hunk 2]
```diff
@@ -35,34 +35,50 @@ module.exports = {
...
+        'static-methods',
+        'lifecycle',
         '/^(on|handle).+$/',
         '/^get.+$/',
         'everything-else',
-        '/^render.+$/',
-        'render',
+        'rendering',
       ],
+      'groups': {
```
Comment (davezuko) [inline frontend/_base.js]: Organized this further into groups.

File: frontend/_base.js [hunk 3]
```diff
@@ -35,34 +35,50 @@ module.exports = {
...
+        ],
+      },
     }],
 
     /**
+     * Compatibility from 1.0.0 -> 2.0.0 migration, mostly new rules
+     * from Airbnb config that conflict with our existing code base.
+     */
+    // Prevent function creation on each render
+    'react/jsx-no-bind': 1, // intended: 2
```
Comment (davezuko) [inline frontend/_base.js]: Talked with @levithomason about this, we _should_ be at a 2 but until we fix occurrences we're just going to keep it as a warning.

**Comment by levithomason** [comment]
has this config been  tested against an app yet?

**Comment by davezuko** [comment]
@levithomason it has a few breaking changes due to mistakes not being caught before, but will be publishing this as a major version and making a lint PR into both apps for the upgrade as we discussed.

**Comment by levithomason** [comment]
:accept:

**Pull request merged**